tests/end_to_end_tests/run_in_remote_machine/__init__.py

```diff
@@ -16,156 +16,194 @@
 This module defines logic that allows to run end-to-end tests inside remote machines such as ec2 instance or docker
 container.
 """
-
+import dataclasses
 import logging
 import pathlib as pl
 import subprocess
-from typing import List, Dict
+from typing import List, Dict, Union
 
 from agent_build_refactored.tools.constants import Architecture
 from agent_build_refactored.tools.run_in_ec2.constants import EC2DistroImage
 
 logger = logging.getLogger(__name__)
 
+_EMBEDDED_OPENSSL_VERSION_NUMBER = 269488447
```

Comment:
Should we add comment in the place where we download + build OpenSSL that this constant in the test needs to be updated if / when OpenSSL version is upgraded there?